frontend/tests/controllers/ContestUsersTest.php

```diff
@@ -551,6 +551,88 @@ public function testContestPracticeWhenOriginalContestHasNotEnded() {
         }
     }
 
+    public function testNextRegisteredContestForUser() {
+        // Create 2 users
+        ['user' => $user1, 'identity' => $identity1] = \OmegaUp\Test\Factories\User::createUser(new \OmegaUp\Test\Factories\UserParams([
+            'username' => 'test_contest_user_1',
+        ]));
+        ['user' => $user2, 'identity' => $identity2] = \OmegaUp\Test\Factories\User::createUser(new \OmegaUp\Test\Factories\UserParams([
+            'username' => 'test_contest_user_2',
+        ]));
+
+        // Get the next registered contest for a user who is not registered in any contest
+        $nextRegisteredContestForUser1 = \OmegaUp\DAO\Contests::getNextRegisteredContestForUser(
+            $identity1
+        );
+        $this->assertNull($nextRegisteredContestForUser1);
+
+        $currentTime = \OmegaUp\Time::get();
+        $timePast =  new \OmegaUp\Timestamp($currentTime - 120 * 60);
+        $timeFuture1 =  new \OmegaUp\Timestamp($currentTime + 60 * 60);
+        $timeFuture2 =  new \OmegaUp\Timestamp($currentTime + 120 * 60);
+        // Get 2 active contests
+        $contest1 = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'title' => 'Contest_1',
+                'startTime' => $timePast,
+                'finishTime' => $timeFuture2,
+            ])
+        );
+        $contest2 = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'title' => 'Contest_2',
+                'startTime' => $timePast,
+                'finishTime' => $timeFuture1,
+            ])
+        );
+        // Get a future contest
+        $contest3 = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'title' => 'Contest_3',
+                'startTime' => $timeFuture1,
+                'finishTime' => $timeFuture2,
+            ])
+        );
+
+        // Add user1 to the active contests
+        \OmegaUp\Test\Factories\Contest::addUser(
+            $contest1,
+            $identity1
+        );
+        \OmegaUp\Test\Factories\Contest::addUser(
+            $contest2,
+            $identity1
+        );
+
+        // Check that the next registered contest for user1 is contest2
```

Comment:
Acá también habrá que indicar por qué motivo el concurso 2 es el que de debería obtener si hay más de un concurso que inician a la misma hora:
```suggestion
        // Check that the next registered contest for user1 is contest2
        // Although both contests started at the same time, contest2 is scheduled
        // to finish sooner, making it the next registered contest.
```